### PR TITLE
Implemented "paying" for travelling followers

### PR DESCRIPTION
**Pull request by NeveHanter**
Player now pays for the following actors when travelling with the exception of the first follower who travels for free, by an addition replaced two versions of getFollowers by getActorsFollowing and getActorsSidingWith methods in MechanicsManager & Actor classes.

### CODE REVIEW

File: apps/openmw/mwgui/travelwindow.cpp [hunk 1]
```diff
@@ -70,14 +70,34 @@ namespace MWGui
...
+            std::string script = follower.getClass().getScript(follower);
+            if (!script.empty() && follower.getRefData().getLocals().getIntVar(script, "stayoutside") == 1)
+                continue;
+
+            if ((follower.getRefData().getPosition().asVec3() - player.getRefData().getPosition().asVec3()).length2() <= 800*800)
+                ++travellingFollowers;
+        }
+
+        // Apply followers cost, in vanilla one follower travels for free
+        price *= travellingFollowers;
```
Comment (MiroslavR) [inline apps/openmw/mwgui/travelwindow.cpp]: Are you sure _price_ should be 0 when the player travels alone? That seems wrong.

Comment (NeveHanter) [inline apps/openmw/mwgui/travelwindow.cpp]: Good catch, fixed, thanks!

File: apps/openmw/mwgui/travelwindow.cpp [hunk 2]
```diff
@@ -70,14 +70,34 @@ namespace MWGui
...
+            std::string script = follower.getClass().getScript(follower);
+            if (!script.empty() && follower.getRefData().getLocals().getIntVar(script, "stayoutside") == 1)
+                continue;
+
+            if ((follower.getRefData().getPosition().asVec3() - player.getRefData().getPosition().asVec3()).length2() <= 800*800)
+                ++travellingFollowers;
+        }
+
+        // Apply followers cost, in vanilla one follower travels for free
+        price *= std::max(1, travellingFollowers);
```
Comment (MiroslavR) [inline apps/openmw/mwgui/travelwindow.cpp]: From my testing, it seems this line should be swapped with the getBarterOffer() line.

**Comment by Gaerzi** [comment]
I feel like this should be subject to a GMST or some other way to make it optional since it diverges from vanilla behavior.

**Comment by MiroslavR** [comment]
> it diverges from vanilla behavior.

What does? AFAIK, this is exactly how vanilla behaves. See https://wiki.openmw.org/index.php?title=Research:Trading_and_Services#Travel_costs.

File: apps/openmw/mwgui/travelwindow.cpp [hunk 3]
```diff
@@ -70,14 +70,34 @@ namespace MWGui
...
+        {
+            MWWorld::Ptr follower = *it;
+
+            std::string script = follower.getClass().getScript(follower);
+            if (!script.empty() && follower.getRefData().getLocals().getIntVar(script, "stayoutside") == 1)
+                continue;
+
+            if ((follower.getRefData().getPosition().asVec3() - player.getRefData().getPosition().asVec3()).length2() <= 800*800)
+                ++travellingFollowers;
+        }
```
Comment (scrawl) [inline apps/openmw/mwgui/travelwindow.cpp]: Please reuse the code that is in ActionTeleport, e.g. by moving it to a function ActionTeleport::getFollowersToTeleport or something like that

**Comment by jeffreyhaines** [comment]
I saw this https://bugs.openmw.org/issues/1501

Is there a ticket for this bug in the tracker?

File: apps/openmw/mwworld/actionteleport.hpp [hunk 4]
```diff
@@ -23,9 +24,12 @@ namespace MWWorld
 
         public:
 
-            ActionTeleport (const std::string& cellName, const ESM::Position& position, bool teleportFollowers);
             ///< If cellName is empty, an exterior cell is assumed.
```
Comment (MiroslavR) [inline apps/openmw/mwworld/actionteleport.hpp]: Please remove the "<" as it's now showing in the generated docs.

Comment (MiroslavR) [inline apps/openmw/mwworld/actionteleport.hpp]: Please remove the "<" as it's now showing in the generated docs.

**Pull request merged**